Applications/VGG/jni/main.cpp

```diff
@@ -83,114 +83,121 @@ createRealDataGenerator(const std::string &directory, unsigned int batch_size,
 
 int main(int argc, char *argv[]) {
 
-  if (argc < 3) {
-    std::cout << "./nntrainer_vgg vgg.ini resource\n";
-    exit(-1);
-  }
-  std::shared_ptr<nntrainer::profile::GenericProfileListener> listener;
+  try {
+    if (argc < 3) {
+      std::cout << "./nntrainer_vgg vgg.ini resource\n";
+      exit(-1);
+    }
+    std::shared_ptr<nntrainer::profile::GenericProfileListener> listener;
 
 #ifdef PROFILE
-  listener = std::make_shared<nntrainer::profile::GenericProfileListener>();
+    listener = std::make_shared<nntrainer::profile::GenericProfileListener>();
 #endif
-  PROFILE_BEGIN(listener);
-
-  seed = time(NULL);
-  srand(seed);
-
-  const std::vector<std::string> args(argv + 1, argv + argc);
-  std::string config = args[0];
-  resource = args[1];
-
-  std::array<UserDataType, 2> user_datas;
-
-  try {
-    user_datas = createRealDataGenerator(resource, batch_size, 1);
-  } catch (const std::exception &e) {
-    std::cerr << "uncaught error while creating data generator! details: "
-              << e.what() << std::endl;
-    return 1;
-  }
-
-  auto &[train_user_data, valid_user_data] = user_datas;
-
-  std::unique_ptr<ml::train::Dataset> dataset_train;
-  try {
-    dataset_train = ml::train::createDataset(
-      ml::train::DatasetType::GENERATOR, trainData_cb, train_user_data.get());
-  } catch (const std::exception &e) {
-    std::cerr << "Error during create train dataset: " << e.what() << std::endl;
-    return 1;
-  }
-
-  std::unique_ptr<ml::train::Dataset> dataset_valid;
-  try {
-    dataset_valid = ml::train::createDataset(
-      ml::train::DatasetType::GENERATOR, validData_cb, valid_user_data.get());
-  } catch (const std::exception &e) {
-    std::cerr << "Error during create valid dataset: " << e.what() << std::endl;
-    return 1;
-  }
-
-  /**
-   * @brief     Neural Network Create & Initialization
-   */
-
-  ModelHandle model;
-  try {
-    model = ml::train::createModel(ml::train::ModelType::NEURAL_NET);
-  } catch (const std::exception &e) {
-    std::cerr << "Error during create model: " << e.what() << std::endl;
-    return 1;
-  }
-
-  try {
-    model->load(config, ml::train::ModelFormat::MODEL_FORMAT_INI);
+    PROFILE_BEGIN(listener);
```

Comment:
nested try-catch is not recommended.

Reply:
Thank you for teaching good coding practices!